File: Leetcode/299. Bulls and Cows.cpp
class Solution {
public:
    string getHint(string secret, string guess) {
        int a[10] = {0},b[10] = {0};
        int matched = 0,cows=0;
        string ans = "";
        for(int i=0;i<secret.length();i++){
            if(secret[i] == guess[i])
            matched++;
            a[secret[i] - 48]++;
            b[guess[i] - 48]++;
        }
        for(int i=0;i<10;i++)
        cows+=min(a[i],b[i]);
        ans = to_string(matched) + "A" + to_string(cows-matched) + "B";
        return ans;
    }
};

File: Leetcode/299. Bulls and Cows.java
class Solution {
    public String getHint(String secret, String guess) {
        Map<Integer, Integer> ct = new HashMap<Integer, Integer>();
        int n = secret.length();
        int bulls = 0;
        int cows = 0;
        for(int i=0;i<n;i++){
            if(secret.charAt(i) == guess.charAt(i)){
                bulls++;
            }else{
                int corr = secret.charAt(i) - '0';
                int inCorr = guess.charAt(i) - '0';
                ct.put(corr, ct.getOrDefault(corr,0) + 1);
                ct.put(inCorr+10, ct.getOrDefault(inCorr+10,0) + 1);
            }
        }
        for(int i=0;i<10;i++){
            cows+= Math.min(ct.getOrDefault(i,0),ct.getOrDefault(i+10,0));
        }
        String ans = Integer.toString(bulls) + "A" + Integer.toString(cows) + "B";
        return ans;
    }
}
